pkg/provider/azure_storageaccount.go

```diff
@@ -578,17 +575,21 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 		if accountOptions.StorageType == StorageTypeBlob {
 			privateEndpointName = privateEndpointName + blobNameSuffix
 		}
-		if err := az.createPrivateEndpoint(ctx, accountName, storageAccount.ID, privateEndpointName, vnetResourceGroup, vnetName, subnetName, location, accountOptions.StorageType); err != nil {
-			return "", "", fmt.Errorf("create private endpoint for storage account(%s), resourceGroup(%s): %w", accountName, vnetResourceGroup, err)
+		var privateEndpointNames []string
+		var rerr error
+		if privateEndpointNames, rerr = az.createPrivateEndpoint(ctx, accountName, storageAccount.ID, privateEndpointName, vnetResourceGroup, vnetName, subnetName, location, accountOptions.StorageType); rerr != nil {
+			return "", "", fmt.Errorf("create private endpoint for storage account(%s), resourceGroup(%s): %w", accountName, vnetResourceGroup, rerr)
 		}
 
-		// Create dns zone group
+		// Create one dns zone group mapping to one vnet and each private endpoint
 		dnsZoneGroupName := accountName + "-dnszonegroup"
 		if accountOptions.StorageType == StorageTypeBlob {
 			dnsZoneGroupName = dnsZoneGroupName + blobNameSuffix
 		}
-		if err := az.createPrivateDNSZoneGroup(ctx, dnsZoneGroupName, privateEndpointName, vnetResourceGroup, vnetName, privateDNSZoneName); err != nil {
-			return "", "", fmt.Errorf("create private DNS zone group - privateEndpoint(%s), vNetName(%s), resourceGroup(%s): %w", privateEndpointName, vnetName, vnetResourceGroup, err)
+		for _, privateEndpointNameOnSubnet := range privateEndpointNames {
```

Comment:
add comments:
create one private endpoint associated with one subnet

Reply:
here is creating a private dns zone group but not private endpoint.

Reply:
I use comment `// Create or update private DNS zone group associated with each private endpoint which associated with one subnet`